src/library/scala/collection/immutable/Queue.scala

```diff
@@ -108,8 +108,14 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
 
   override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
     if (bf eq Queue.ReusableCBF) {
-      val thatQueue = that.asInstanceOf[Queue[B]]
-      new Queue[B](thatQueue.in ++ (thatQueue.out reverse_::: this.in), this.out).asInstanceOf[That]
+      val newIn =
+        if (that.isInstanceOf[Queue[_]]) {
+          val thatQueue: Queue[B] = that.asInstanceOf[Queue[B]]
+          thatQueue.in ++ (thatQueue.out reverse_::: this.in)
+        } else {
+          (new ListBuffer[B] ++= that.seq).prependToList(this.in)
```

Comment:
Yet another app-breaking bug 😞 There is no such comprehensive, constructive testing for the Scala compiler or stdlib. All there is is regression tests; the entire Scala community is punished for every contributor's mistakes. until the next version, for months at a time. @Ichoran is working on it at the moment for collections; for now, I have the fix and can avoid actually reversing `that.seq`. Thanks a lot for the test-cases and report.